### ISSUE TITLE
Holo-Collector AI – Was brauche ich für RunPod + On-Device?

### ISSUE THREAD

**Issue by gprojects-ch** [gprojects-ch/holo-collector-ai#8]
# Holo-Collector AI – Was brauche ich für RunPod + On-Device?

### 1.1 Neues Repo / Ordner
- `holo-collector-ai/`
  - `data/` → Bilder + Labels
  - `configs/` → Modell- und Training-Configs
  - `scripts/` → Trainings-/Eval-/Export-Skripte
  - `models/` → gespeicherte Checkpoints + exportierte .tflite

### 1.2 Tools / Stack
- Python (z.B. 3.10+)
- Deep-Learning-Framework:
  - z.B. PyTorch + YOLO (z.B. v8/11) **oder**
  - TensorFlow/Keras, wenn du TFLite direkt nutzen willst
- RunPod-Template/Machine mit GPU (z.B. 1x 3090/4090 o.ä.)

### 1.3 Basis-Skripte (am Anfang nur leer / TODO)
- `scripts/train.py` → trainiert das Modell
- `scripts/eval.py` → evaluiert (mAP, Accuracy)
- `scripts/export_tflite.py` → exportiert Modell fürs Handy (`.tflite`)
- `configs/minifig.yaml` → z.B. YOLO-Config mit Klassenliste & Pfaden

> Ziel: Du hast ein **sauberes Trainings-Projekt**, das du auf RunPod klonen und direkt starten kannst.

---

### 2.1 Bilder sammeln
- Eigene Fotos:
  - Deine Star-Wars-Minifigs und Sets
  - Verschiedene Hintergründe, Licht, Perspektiven
- Ordnerstruktur z.B.:
  - `data/raw/` → unbearbeitete Fotos
  - `data/processed/` → zugeschnittene/optimierte Bilder

### 2.2 Label-Tool auswählen
- Ein Tool für Bounding Boxes:
  - z.B. Label Studio, CVAT, Roboflow, makesense.ai etc.
- Brauchst:
  - **Klassenliste** (zuerst Top 100–300 Minifigs)
  - Export im z.B. YOLO-Format (TXT + Bilder) oder COCO-JSON

### 2.3 Klassen-Definition
- Datei z.B. `configs/classes.yaml` oder `data/classes.txt`:
  - `0 sw0001`
  - `1 sw0002`
  - `2 sw0554`
  - `3 sw0700`
  - ...
- Wichtig:
  - Diese IDs müssen **stabil** bleiben, weil du sie später im Handy brauchst (Mapping von Modell → Minifig-ID in deiner DB)

### 2.4 Minimale Datenmenge für Start
- Pro Klasse:
  - ideal: 50–100 Bilder (MVP geht auch mit weniger, Hauptfiguren zuerst)
- Mischung aus:
  - einzelne Figur
  - mehrere Figuren im Bild
  - verschiedene Posen / teil-verdeckt / schräg

> Ziel: Du hast **einen ersten kleinen, aber sauberen Datensatz** + Klassenliste.

---

## 3. Mobile App (Flutter) – On-Device Interface vorbereiten

Auch wenn das Modell noch nicht da ist, brauchst du in der App:

### 3.1 Datenstruktur für Erkennungen
```dart
class MinifigDetection {
  final int classId;
  final String externalId; // z.B. "sw0700"
  final String name;       // kommt aus deiner DB
  final Rect bbox;
  final double confidence;

  MinifigDetection({
    required this.classId,
    required this.externalId,
    required this.name,
    required this.bbox,
    required this.confidence,
  });
}
